app/models/document/paginated_timeline.rb

```diff
@@ -89,4 +95,30 @@ def prev_page
       false
     end
   end
+
+private
+
+  def next_page_entries
+    if next_page
+      Document::PaginatedTimelineQuery.new(document:, page: next_page, only:).raw_entries
+    end
+  end
+
+  def host_content_update_events
+    return [] if query.raw_entries.empty?
+
+    @host_content_update_events ||= HostContentUpdateEvent.all_for_date_window(
+      document:,
+      from: date_window.last,
+      to: date_window.first,
+    )
+  end
+
+  def date_window
+    @date_window ||= begin
+      start = page == 1 ? Time.zone.now : query.raw_entries.first.created_at
+      ends = next_page_entries ? next_page_entries.first.created_at : query.raw_entries.last.created_at
```

Comment:
The `next_page` logic is surprising - can you add to the commit detail what you've added to the PR description?

```
This is complicated slightly by the fact that the history is paginated. To compensate for this, we are having to "look ahead" to the next page to get the date of the next event to give us a window to query Publishing API for events.
```

Better yet, perhaps just handle the first page in this first commit, and then add the "next page" logic in a subsequent commit, rather than doing it all at once.